### ISSUE TITLE
GeckoViewFetchTestCases failing: get200WithDefaultHeaders, get200WithCookiePolicy

### ISSUE THREAD

**Issue by isabelrios** [mozilla-mobile/android-components#6407]
There are two tests failing in all PRs for all APIs:

-get200WithDefaultHeaders: [Logs](https://console.firebase.google.com/project/moz-android-components-230120/testlab/histories/bh.9f526cd30412cc12/matrices/6988824370638142580/executions/bs.ad34f9b285c9f859/testcases/12)
-get200WithCookiePolicy: [Logs](https://console.firebase.google.com/project/moz-android-components-230120/testlab/histories/bh.9f526cd30412cc12/matrices/6988824370638142580/executions/bs.ad34f9b285c9f859/testcases/19)

These tests run as part as the current [test-ui task](https://github.com/mozilla-mobile/android-components/runs/537031958)See task at the very bottom. Although the tests are failing, the task is green. That will be fixed in PR #6279, task will be red if tests fail.

**Comment by pocmo** [comment]
### get200WithDefaultHeaders

We expect 6 headers to be set. But apparently there are 9.

### get200WithCookiePolicy

Using Request.CookiePolicy.INCLUDE doesn't seem to include the cookies 🤔

**Comment by pocmo** [comment]
The additional headers that GeckoView sends are:

```
Sec-Fetch-Dest: empty
Sec-Fetch-Mode: no-cors
Sec-Fetch-Site: cross-site
```

**Comment by pocmo** [comment]
https://bugzilla.mozilla.org/show_bug.cgi?id=1626335

**Comment by pocmo** [comment]
GeckoView bug is fixed and test was re-enabled in 5ed2f7c1566688e229561f690586571b390bd33a
